test/integration/cdi/example-driver/kubeletplugin/nonblockinggrpcserver.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package app does all of the work necessary to configure and run a
+// Kubernetes app process.
+package server
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"sync"
+
+	"google.golang.org/grpc"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+)
+
+// NonBlocking server
+type nonBlockingGRPCServer struct {
+	wg      sync.WaitGroup
+	server  *grpc.Server
+	cleanup func()
+}
+
+func (s *nonBlockingGRPCServer) Start(endpoint string, ns drapbv1.NodeServer) {
+
+	s.wg.Add(1)
```

Comment:
Something seems to haven gotten lost as you copied this code around: where is the corresponding `s.wg.Done()`?

Reply:
Did I?
https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/pkg/hostpath/server.go#L43
Maybe,, the example I got from was not really good option...

Reply:
But, I studed about waitinggroup a bit, and I understand your point.!

Reply:
I don't know where it went of the rails, but `s.wg.Add(1)` followed by `s.wg.Done()` makes no sense. It becomes a no-op. The entire wait group serves no purpose anymore in the current revision of the code.

Perhaps it is not needed (always spawn goroutine, never wait for it), but then the entire code could be simpler.